src/test/java/com/github/martynagil/drugstoremanagement/service/ComplaintServiceTest.java

```diff
@@ -3,137 +3,246 @@
-			"shop name",
-			address
-	);
-	private final List<TransactionEntry> transactionEntries = new ArrayList<>();
-	private final Transaction transaction = new Transaction(
-			LocalDateTime.now(),
-			shop,
-			transactionEntries
-	);
-	private ComplaintUpdateDto complaintUpdateDto = new ComplaintUpdateDto(
-			ComplaintStatus.ACCEPTED
-	);
-	private Complaint complaint = new Complaint(
-			LocalDate.parse("2021-01-13"),
-			"reason",
-			product,
-			transaction
-	);
+	@Mock
 	private ComplaintRepository complaintRepository;
+	@Mock
 	private ProductRepository productRepository;
+	@Mock
 	private TransactionRepository transactionRepository;
-	private ComplaintService complaintService;
-
-	@BeforeEach
-	void setUp() {
-		complaintRepository = mock(ComplaintRepository.class);
-		productRepository = mock(ProductRepository.class);
-		transactionRepository = mock(TransactionRepository.class);
-		complaintService = new ComplaintService(complaintRepository, productRepository, transactionRepository);
-	}
+	@InjectMocks
+	private ComplaintService complaintService = new ComplaintService(complaintRepository, productRepository, transactionRepository);
 
 	@Test
 	void shouldAddComplaintWhenItDoesNotExist() {
-		when(complaintRepository.existsByTransactionIdAndProductIdAndComplaintStatus(any(), any(), any()))
-				.thenReturn(false);
-		when(productRepository.findById(any()))
-				.thenReturn(Optional.of(product));
-		when(transactionRepository.findById(any()))
-				.thenReturn(Optional.of(transaction));
+		complaintDoesNotExistForAdding();
+		findProductInRepository();
+		findTransactionInRepository();
 
-		complaintService.addComplaint(complaintDto);
+		complaintService.addComplaint(complaintDto());
 
 		verify(complaintRepository).save(any());
 	}
 
 	@Test
 	void shouldNotAddComplaintWhenIdDoesExist() {
-		when(complaintRepository.existsByTransactionIdAndProductIdAndComplaintStatus(any(), any(), any()))
-				.thenReturn(true);
+		complaintExistsForAdding();
 
 		assertThrows(
 				ComplaintAlreadyExistsException.class,
-				() -> complaintService.addComplaint(complaintDto)
+				() -> complaintService.addComplaint(complaintDto())
 		);
-
 		verify(complaintRepository, never()).save(any());
 	}
 
 	@Test
 	void shouldNotUpdateWhenComplaintDoesNotExists() {
-		when(complaintRepository.findById(any()))
-				.thenReturn(Optional.empty());
+		complaintDoesNotExistsForUpdating();
 
 		assertThrows(
 				EntityNotFoundException.class,
-				() -> complaintService.updateComplaint(complaintUpdateDto, 1L)
+				() -> complaintService.updateComplaint(complaintUpdateToAcceptedDto(), 1L)
 		);
-
 		verify(complaintRepository, never()).save(any());
 	}
 
 	@Test
-	void shouldUpdateWhenComplaintExists() {
-		when(complaintRepository.findById(any()))
+	void shouldUpdateWhenSubmittedComplaintExists() {
+		complaintExistsForUpdating(complaint());
+		Complaint complaint = complaintRepository.findById(1L)
+				.orElseThrow(EntityNotFoundException::new);
```

Comment:
czemu coś wyciągasz z mocka podczas testów? dziwne to

Reply:
nie rozumiem